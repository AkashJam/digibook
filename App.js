import { StyleSheet, View } from "react-native";
import AppLoading from "expo-app-loading";
import { useFonts } from "expo-font";
import { HomePage, TestPage } from "./src/screens"

export default function App() {

  // console.log("Screen Dimensions", Dimensions.get('window'))
  // console.log('statusBarHeight: ', StatBar.currentHeight);

  let [fontsLoaded] = useFonts({
    "Caviar_Dreams_Bold": require("./assets/fonts/CaviarDreams_Bold.ttf"),
    "Fredoka_Regular": require("./assets/fonts/Fredoka-Regular.ttf"),
  });

  if (!fontsLoaded) {
    return <AppLoading />;
  } else {
    return (  
      // <TestPage />
      <HomePage />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 32,
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
