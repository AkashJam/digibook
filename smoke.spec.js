import React from "react";
import renderer from "react-test-renderer";
import App from "./App";
// import { useFonts } from "expo-font"

jest.mock("expo-font", () => {
  // const actualFonts = jest.requireActual('expo-fonts');
  return {
    // ...actualFonts,
    useFonts: () => {
      return [{ fontsLoaded: true }];
    },
  };
});

jest.mock("expo-constants", () => {
  const Constants = { isDevice: false };
  return Constants;
});

jest.mock("react-native-gesture-handler", () => {
  const { View } = require("react-native");
  return {
    GestureHandlerRootView: View,
  };
});

jest.mock("react-native-root-toast", () => {
  const { View } = require("react-native");
  return {
    Toast: View,
  };
});

jest.mock("react-native-maps", () => {
  const { View } = require("react-native");
  return {
    MapView: View,
    Marker: View,
  };
});

// describe("truth", () => {
//   it("is true", () => {
//     expect(true).toEqual(true);
//   });
// });

// it("checks if Async Storage is used", async () => {
//   await asyncOperationOnAsyncStorage();

//   expect(AsyncStorage.getItem).toBeCalledWith("DN_userlog");
// });

it("renders without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

it("test against snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});


