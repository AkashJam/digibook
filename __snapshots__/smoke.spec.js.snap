// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test against snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0"
>
  <div
    className="css-view-1dbjc4n r-flex-13awgt0"
  >
    <div
      className="css-view-1dbjc4n r-flex-13awgt0"
      style={
        Object {
          "backgroundColor": "rgba(242,242,242,1.00)",
        }
      }
    >
      <div
        className="css-view-1dbjc4n r-flex-13awgt0"
      >
        <div
          className="css-view-1dbjc4n r-bottom-1p0dtai r-left-1d2f490 r-pointerEvents-12vffkv r-position-u8s1d r-right-zchlnj r-top-ipm5af"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <div
            className="css-view-1dbjc4n r-bottom-1p0dtai r-left-1d2f490 r-pointerEvents-12vffkv r-position-u8s1d r-right-zchlnj r-top-ipm5af"
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              className="css-view-1dbjc4n r-bottom-1p0dtai r-left-1d2f490 r-pointerEvents-12vffkv r-position-u8s1d r-right-zchlnj r-top-ipm5af"
            />
            <div
              className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
              id="animatedComponent"
            >
              <div
                className="css-view-1dbjc4n r-flex-13awgt0"
                id="animatedComponent"
              >
                <div
                  className="css-view-1dbjc4n r-minHeight-2llsf"
                  style={
                    Object {
                      "backgroundColor": "rgba(242,242,242,1.00)",
                    }
                  }
                >
                  <div
                    className="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-1d5kdc7"
                  >
                    <div
                      className="css-view-1dbjc4n r-flex-13awgt0"
                    >
                      <div
                        className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-i7nevx r-height-1pi2tsx r-paddingHorizontal-1x7blby r-paddingTop-1gingfq"
                      >
                        <div
                          className="css-text-901oao r-color-18or775"
                          dir="auto"
                          style={
                            Object {
                              "fontFamily": "Caviar_Dreams_Bold",
                              "fontSize": "24px",
                              "paddingBottom": "4%",
                              "paddingLeft": "4%",
                              "paddingRight": "4%",
                              "paddingTop": "4%",
                            }
                          }
                        >
                          Digital Notebook
                        </div>
                        <input
                          autoCapitalize="sentences"
                          autoComplete="on"
                          autoCorrect="on"
                          className="css-textinput-11aywtz r-placeholderTextColor-1o1bpf7"
                          dir="auto"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onSelect={[Function]}
                          placeholder="Username"
                          readOnly={false}
                          spellCheck={true}
                          style={
                            Object {
                              "borderBottomColor": "rgba(0,166,251,1.00)",
                              "borderLeftColor": "rgba(0,166,251,1.00)",
                              "borderRightColor": "rgba(0,166,251,1.00)",
                              "borderTopColor": "rgba(0,166,251,1.00)",
                            }
                          }
                          type="text"
                          value=""
                        />
                        <input
                          autoCapitalize="sentences"
                          autoComplete="on"
                          autoCorrect="on"
                          className="css-textinput-11aywtz r-placeholderTextColor-1o1bpf7"
                          dir="auto"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onSelect={[Function]}
                          placeholder="Password"
                          readOnly={false}
                          spellCheck={true}
                          style={
                            Object {
                              "borderBottomColor": "rgba(0,166,251,1.00)",
                              "borderLeftColor": "rgba(0,166,251,1.00)",
                              "borderRightColor": "rgba(0,166,251,1.00)",
                              "borderTopColor": "rgba(0,166,251,1.00)",
                            }
                          }
                          type="password"
                          value=""
                        />
                        <div
                          aria-disabled={true}
                          className="css-view-1dbjc4n r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onKeyDown={[Function]}
                          style={
                            Object {
                              "WebkitTransitionDuration": "0s",
                              "backgroundColor": "rgba(246,240,237,1.00)",
                              "transitionDuration": "0s",
                            }
                          }
                          tabIndex="-1"
                        >
                          <div
                            className="css-text-901oao r-fontFamily-1cdzzfu r-fontSize-1i10wst"
                            dir="auto"
                            style={
                              Object {
                                "color": "rgba(54,54,53,1.00)",
                              }
                            }
                          >
                            LOGIN
                          </div>
                        </div>
                        <div
                          aria-disabled={true}
                          className="css-view-1dbjc4n r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onKeyDown={[Function]}
                          style={
                            Object {
                              "WebkitTransitionDuration": "0s",
                              "backgroundColor": "rgba(246,240,237,1.00)",
                              "transitionDuration": "0s",
                            }
                          }
                          tabIndex="-1"
                        >
                          <div
                            className="css-text-901oao r-fontFamily-1cdzzfu r-fontSize-1i10wst"
                            dir="auto"
                            style={
                              Object {
                                "color": "rgba(54,54,53,1.00)",
                              }
                            }
                          >
                            REGISTER
                          </div>
                        </div>
                        <div
                          className="css-text-901oao r-color-18or775 r-fontFamily-1cdzzfu r-fontSize-1i10wst r-paddingHorizontal-cf760m r-paddingVertical-1crsq4q r-textAlign-q4m81j"
                          dir="auto"
                        >
                          First time? Get started with just a username and password
                        </div>
                      </div>
                    </div>
                    <div
                      className="css-view-1dbjc4n r-pointerEvents-12vffkv"
                      id="animatedComponent"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-left-1d2f490 r-pointerEvents-12vffkv r-position-u8s1d r-right-zchlnj r-top-ipm5af"
        id="animatedComponent"
        style={
          Object {
            "height": "64px",
            "zIndex": 1,
          }
        }
      />
    </div>
  </div>
</div>
`;
